third_party/amd/lib/TritonAMDGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -27,6 +28,56 @@ using ::mlir::triton::AMD::ISAFamily;
 using ::mlir::triton::gpu::getTotalElemsPerThread;
 namespace {
 
+std::optional<LLVM::AtomicBinOp> matchAtomicOp(RMWOp atomicOp) {
+  static const std::unordered_map<RMWOp, LLVM::AtomicBinOp> atomicMap = {
```

Comment:
A static `switch` statement would do? We don't need the overhead to build an `static` unordered map?